packages/integrations/mdx/src/plugins.ts

```diff
@@ -74,7 +74,7 @@ function getRehypePlugins(mdxOptions: MdxOptions): PluggableList {
 		}
 	}
 
-	rehypePlugins.push(...mdxOptions.rehypePlugins);
+	rehypePlugins.push(...mdxOptions.rehypePlugins, rehypeImageToComponent);
```

Comment:
Hmm, that's a bit annoying and makes this breaking. Previously, it was possible for a rehype plugin to change the options passed to the component, whereas right now it's just impossible. Wonder if anyone depended on that.

Reply:
From my understanding (don't hesitate to correct me), the properties on a node could be unrelated with HTML attributes. So I think it is still possible to pass options to the `Image` component. Using a rehype plugin (or even a remark plugin), the consumer could add the options as properties on images. The properties are then mapped to attributes in `rehype-images-to-component` (only `src` is hard coded) so the `Image` component will receive those options. However, the consumer needs to target `img` instead of `astro-image` as node name. So, yes I see the breaking change now (and the `major` detected by changeset is justified).

However, I don't see how we could fix the linked issue without this change... From what I saw with a quick search (so I certainly missed some use cases), there are essentially two use cases:
* those who replicate existing rehype plugins to handle `astro-image` nodes
* those who duplicate their logic to handle `astro-image` in addition to `img`

So I think this change is beneficial (if my understanding is correct). But it is definitely a breaking change.

That said, perhaps we need to check the correct formatting of the options if they exist. In comparison with the remark plugin, I had to add a special case for `widths` because it was a string instead of an array (based on existing tests) and I added `decodeUri` on `src` to handle paths with spaces.

Reply:
Yes previously a rehype plugin could set width to be 100 like this: 

```js
if (node.name === 'astro-image') {
node.attributes.push({
	name: 'width',
	type: 'mdxJsxAttribute',
	value: '100',
  });
}
```

now after this change it will be:

```js
if (node.tagName === 'img') {
node.properties = {
	...node.properties,
	width: '100',
  };
}
```

Reply:
yes previously a rehype plugin could set width to 100 like this:

```js
if (node.name === 'astro-image') {
	node.attributes.push({
		name: 'width',
		type: 'mdxJsxAttribute',
		value: '100',
	});
}
```

now after this change it will be like this:

```js
if (node.tagName === 'img') {
	node.properties = {
		...node.properties,
		width: '100',
	};
}
```